### PR TITLE
COTR-75 Made enemies spawn within radius

### PR DESCRIPTION
**Pull request by Arnelys**
Made enemies spawn within a set radius of 400 units. Made sure enemies spawn within boundaries of the map.

Also increased timer wait time due to the fact that with the enemies spawning much closer to the player, the game gets too hard too quick.

### CODE REVIEW

File: Scenes/Globals/Globals.cs [hunk 1]
```diff
@@ -161,6 +161,20 @@ public static float GetRandomFloat(float min, float max)
         return (float)scaled;
     }
 
+    /// <summary>
+    /// Method to get a random position within a set radius
+    /// </summary>
+    /// <param name="playerPosition">Player coordinates</param>
+    /// <param name="limit">Radius liimt</param>
+    /// <returns>Vector2 object with position coordinates</returns>
+	public static Vector2 GetRandomPositionWithinRadius(Vector2 playerPosition, int limit)
```
Comment (Elijuszek) [inline Scenes/Globals/Globals.cs:170]: Change variable name `playerposition` to `globalPosition`

File: Scenes/Globals/Globals.cs [hunk 2]
```diff
@@ -161,6 +161,20 @@ public static float GetRandomFloat(float min, float max)
         return (float)scaled;
     }
 
+    /// <summary>
+    /// Method to get a random position within a set radius
+    /// </summary>
+    /// <param name="playerPosition">Player coordinates</param>
+    /// <param name="limit">Radius liimt</param>
```
Comment (Elijuszek) [inline Scenes/Globals/Globals.cs:168]: Mistake in summary "limit"

File: Scenes/Utility/EnemySpawner.cs [hunk 3]
```diff
@@ -18,6 +18,8 @@ public partial class EnemySpawner : Node2D
 
     private int time; //variable to keep count of the time that has passed since spawning started
 
+    private int enemySpawnDistanceLimit = 400; //variable to change how far from the player enemies should spawn
```
Comment (Elijuszek) [inline Scenes/Utility/EnemySpawner.cs:21]: Make this variable Export

File: Scenes/Utility/EnemySpawner.cs [hunk 4]
```diff
@@ -47,15 +49,19 @@ public void Initialize(Vector2 vector, Player player)
...
-    /// <returns>2D vector to reporesent the position of the spawn point</returns>
-    private Vector2 GetRandomPosition()
+    /// <returns>Vector2 object with position coordinates</returns>
+    private Vector2 GetEnemSpawnPosition()
     {
-        float x = Globals.GetRandomFloat(limitLeft, limitRight);
-        float y = Globals.GetRandomFloat(limitTop, limitBottom);
+        Vector2 position = Globals.GetRandomPositionWithinRadius(target.Position, enemySpawnDistanceLimit);
+
+        while (position.Y < limitTop || position.Y > limitBottom || position.X < limitLeft || position.X > limitRight)
```
Comment (Elijuszek) [inline Scenes/Utility/EnemySpawner.cs:59]: Instead of using while use `Math.Clamp(min, max)`

**Review by Elijuszek — CHANGES_REQUESTED** [review]

File: Scenes/Globals/Globals.cs (hunk 1 above)
Comment (Arnelys) [reply Scenes/Globals/Globals.cs:170]: Done

File: Scenes/Globals/Globals.cs (hunk 2 above)
Comment (Arnelys) [reply Scenes/Globals/Globals.cs:168]: Done

File: Scenes/Utility/EnemySpawner.cs (hunk 3 above)
Comment (Arnelys) [reply Scenes/Utility/EnemySpawner.cs:21]: Done

File: Scenes/Utility/EnemySpawner.cs (hunk 4 above)
Comment (Arnelys) [reply Scenes/Utility/EnemySpawner.cs:59]: Done

File: Scenes/Globals/Globals.cs [hunk 5]
```diff
@@ -161,6 +161,20 @@ public static float GetRandomFloat(float min, float max)
         return (float)scaled;
     }
 
+    /// <summary>
+    /// Method to get a random position within a set radius
+    /// </summary>
+    /// <param name="playerPosition">Player coordinates</param>
```
Comment (Elijuszek) [inline Scenes/Globals/Globals.cs:167]: change this comment according to variable name

File: Scenes/Utility/EnemySpawner.cs [hunk 6]
```diff
@@ -18,6 +18,8 @@ public partial class EnemySpawner : Node2D
 
     private int time; //variable to keep count of the time that has passed since spawning started
 
+    [Export] public int enemySpawnDistanceLimit = 400; //variable to change how far from the player enemies should spawn
```
Comment (Elijuszek) [inline Scenes/Utility/EnemySpawner.cs:21]: Add summary instead of basic comment for every property or export variable

**Review by Elijuszek — CHANGES_REQUESTED** [review]

File: Scenes/Globals/Globals.cs (hunk 5 above)
Comment (Arnelys) [reply Scenes/Globals/Globals.cs:167]: Done

File: Scenes/Utility/EnemySpawner.cs (hunk 6 above)
Comment (Arnelys) [reply Scenes/Utility/EnemySpawner.cs:21]: Done

**Review by Elijuszek — APPROVED** [review]
LGTM!

**Pull request merged**